### PR TITLE
move smartWallet/Inter integration to inter-protocol

### PR DESCRIPTION
**Pull request by turadg**
## Description

Refactorings from https://github.com/Agoric/agoric-sdk/pull/6537 to support https://github.com/Agoric/agoric-sdk/issues/4885

### Security Considerations

--
### Documentation Considerations

--
### Testing Considerations

--

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
I don't have the context to understand what this is doing. @dckc  can you review it?  Otherwise I'll  have to ask questions about what's going on.

File: packages/inter-protocol/test/smartWallet/contexts.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { deeplyFulfilledObject } from '@agoric/internal';
```
Comment (dckc) [inline packages/inter-protocol/test/smartWallet/contexts.js:2]: Is this file really new? or was it moved?

File: packages/inter-protocol/test/smartWallet/devices.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// @ts-check
```
Comment (dckc) [inline packages/inter-protocol/test/smartWallet/devices.js:1]: likewise... is this file really new?

File: packages/inter-protocol/test/smartWallet/supports.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
+// @ts-check
```
Comment (dckc) [inline packages/inter-protocol/test/smartWallet/supports.js]: likewise... is this file really new?

**Review by dckc — COMMENTED** [review]
I could use some help reading the diff. And some help understanding how this helps with checkJs.

File: packages/inter-protocol/test/smartWallet/contexts.js (hunk 1 above)
Comment (turadg) [reply packages/inter-protocol/test/smartWallet/contexts.js:2]: For your three "is it really new" questions: no, they're not new but they can't be moved either. They're copies.

These files were moved, but the supporting files are still needed in the smart-wallet package.
Screen Shot 2022-11-04 at 11 33 00 AM

**Comment by turadg** [comment]
> I could use some help reading the diff. And some help understanding how this helps with checkJs.

I'll try to huddle you. It helps with `checkJs` by cleaning up the dependency graph, making less stray JS that `smart-wallet` tries to typecheck

File: packages/inter-protocol/test/smartWallet/supports.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+export { ActionType };
+
+/**
+ * @param {object} kit
+ * @param {Brand<'nat'>} kit.brand
+ * @param {Issuer<'nat'>} kit.issuer
+ * @param {Mint<'nat'>} [kit.mint]
+ */
+export const withAmountUtils = kit => {
```
Comment (dckc) [inline packages/inter-protocol/test/smartWallet/supports.js]: do we have this in inter already?

File: packages/inter-protocol/test/smartWallet/supports.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+/**
+ * @param {bigint} value
+ * @param {{
+ *   feeMintAccess: ERef<FeeMintAccess>,
+ *   zoe: ERef<ZoeService>,
+ * }} powers
+ * @returns {Promise<Payment>}
+ */
+export const mintCentralPayment = async (
```
Comment (dckc) [inline packages/inter-protocol/test/smartWallet/supports.js]: this looks familiar; is there another one of these in inter-protocol?

File: packages/smart-wallet/package.json [hunk 6]
```diff
@@ -19,21 +19,19 @@
     "@agoric/cosmic-proto": "^0.2.1",
     "@agoric/cosmic-swingset": "^0.39.2",
     "@agoric/deploy-script-support": "^0.9.4",
-    "@agoric/inter-protocol": "^0.13.1",
```
Comment (dckc) [inline packages/smart-wallet/package.json:22]: aha. I see. we're getting rid of this dependency. nice.

**Review by dckc — APPROVED** [review]
nice to get rid of the dependency on inter-protocol from smart-wallet

**Comment by turadg** [comment]
@Mergifyio refresh

**Pull request merged**